test/bats/test.bats

```diff
@@ -91,16 +113,6 @@ SLEEP_TIME=1
   wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl get k8suniquelabel.constraints.gatekeeper.sh ns-gk-label-unique -o yaml | grep enforced"
 
   run kubectl apply -f ${BATS_TESTS_DIR}/bad/no_dupe_ns_2.yaml
-  assert_match 'denied the request' "$output"
+  wait_for_process $WAIT_TIME $SLEEP_TIME assert_match 'denied the request' "$output"
```

Comment:
I'm not sure this makes up for the lack of check for sync complete. I believe `kubectl apply` will only send a change to the API server if there is a diff.

Reply:
@maxsmythe moved this to the sync metrics so we are also covering that. let me know what you think

Reply:
Running a `wait_for_process` over the assertion won't change the result, so that can be dropped.

Re: waiting on the sync metric... LGTM